broker/src/session/session.erl

```diff
@@ -173,50 +173,33 @@ ready({dial, _, _, QueuedCall = #queued_call{address = undefined}}, _From, State
 ready({dial, RealBroker, Channel, QueuedCall}, _From, State = #state{session_id = SessionId, resume_ptr = ResumePtr}) ->
   lager:info("Session (~p) dial", [SessionId]),
 
-  NewSession = case State#state.session of
-    undefined ->
-      CallLog = call_log_srv:new(SessionId, call_log:find(QueuedCall#queued_call.call_log_id)),
-      Contact = get_contact(QueuedCall#queued_call.project_id, QueuedCall#queued_call.address, QueuedCall#queued_call.call_log_id),
-      Session = QueuedCall:start_session(),
-
-      poirot:add_meta([
-        {address, QueuedCall#queued_call.address},
-        {project_id, QueuedCall#queued_call.project_id},
-        {call_log_id, QueuedCall#queued_call.call_log_id},
-        {channel_id, Channel#channel.id},
-        {channel_name, Channel#channel.name}
-      ]),
-
-      Session#session{
-        session_id = SessionId,
-        channel = Channel,
-        call_log = CallLog,
-        contact = Contact
-      };
-
-    Session ->
-      CallLog = Session#session.call_log,
-      Session#session{queued_call = QueuedCall, address = QueuedCall#queued_call.address}
-  end,
+  NewSession = #session{call_log = CallLog, flow = Flow} = prepare_session(QueuedCall, Channel, State),
 
-  % Don't the started_at if we are resuming an existing session
+  %% Don't set the started_at if we are resuming an existing session
```

Comment:
Heh, that's all Emacs (or rather erlang-mode) merit which kept indenting the line to the far right.